Source/Core/Common/QoSSession.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Common/QoSSession.h"
+#include "Core/ConfigManager.h"
+
+#if defined(_WIN32)
+#include <Qos2.h>
+#pragma comment(lib, "qwave")
+#endif
+
+namespace Common
+{
+#if defined(_WIN32)
+QoSSession::QoSSession(ENetPeer* peer, int tos_val) :
+  m_peer(peer)
+{
+  constexpr QOS_VERSION ver = {1, 0};
+
+  if (!QOSCreateHandle(&ver, &m_qos_handle))
+    return;
+
+  sockaddr_in sin = { };
+
+  sin.sin_family = AF_INET;
+  sin.sin_port = ENET_HOST_TO_NET_16(peer->host->address.port);
+  sin.sin_addr.s_addr = peer->host->address.host;
+
+  if (QOSAddSocketToFlow(m_qos_handle, peer->host->socket, reinterpret_cast<PSOCKADDR>(&sin),
+                          QOSTrafficTypeControl, QOS_NON_ADAPTIVE_FLOW, &m_qos_flow_id))
+  {
+    // We shift the complete ToS value by 3 to get rid of the 3 bit ECN field
+    DWORD dscp = tos_val >> 3;
```

Comment:
This is performing a sign-conversion without an explicit `static_cast` (signed result being assigned to an unsigned value).